### PR TITLE
build: Add Jenkins job to mocha suite

### PR DESCRIPTION
**Pull request by ewanharris**
Doing so might help reduce the Windows build breakage from migrating tests from the main repo

This wont do anything til appcelerator/jenkins_dsl/pull/74 is potentially merged

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+// Variables we can change
+def nodeVersion = '8.9.1' // NOTE that changing this requires we set up the desired version on jenkins master first!
+def npmVersion = 'latest' // We can change this without any changes to Jenkins. 5.7.1 is minimum to use 'npm ci'
+
+def unitTests(os, nodeVersion, npmVersion, testSuiteBranch, target) {
+	try {
+		nodejs(nodeJSInstallationName: "node ${nodeVersion}") {
+			ensureNPM(npmVersion)
+			sh 'npm ci'
```
Comment (sgtcoolguy) [inline Jenkinsfile]: bat for windows?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+			} // dir('scripts')
+		} // nodejs
+	} finally {
+		deleteDir()
+	}
+}
+
+// Wrap in timestamper
+timestamps {
+	node('git && osx') {
```
Comment (sgtcoolguy) [inline Jenkinsfile]:
```groovy
node('git && (linux || osx)') {
```

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+			stage('Lint') {
+				ensureNPM(npmVersion)
+				// Install dependencies
+				timeout(5) {
+					sh 'npm ci'
+				}
+				sh 'npm test'
+			}
+		}
+	stage('Test') {
```
Comment (sgtcoolguy) [inline Jenkinsfile]: Looks like you're missing a } to close the node scope here before starting the Test stage

File: Jenkinsfile (hunk 1 above)
Comment (ewanharris) [reply Jenkinsfile]: Gah of course, I was thinking of using the shell thing you look to be adding to the pipeline library

**Comment by ewanharris** [comment]
@sgtcoolguy Thanks for finishing this off 😺

**Pull request merged**